pytest/tests/test_contract_update.py

```diff
@@ -11,146 +11,125 @@
 from common_lib.constants import TGAS
 
 sys.path.append(str(pathlib.Path(__file__).resolve().parents[1]))
 from common_lib import shared
-from common_lib.contracts import V0_CONTRACT_PATH, COMPILED_CONTRACT_PATH, MIGRATE_CURRENT_CONTRACT_PATH, V1_CONTRACT_PATH, UpdateArgsV0, UpdateArgsV1, ConfigV1
-
-
-# todo: migration tests
-@pytest.mark.skip
-@pytest.mark.parametrize(
-    "initial_contract_path,update_args",
-    [
-        pytest.param(V1_CONTRACT_PATH,
-                     UpdateArgsV0(COMPILED_CONTRACT_PATH),
-                     id="update v1 to current"),
-        #pytest.param(COMPILED_CONTRACT_PATH,
-        #             UpdateArgsV1(code_path=MIGRATE_CURRENT_CONTRACT_PATH),
-        #             id="update current code"),
-        #pytest.param(COMPILED_CONTRACT_PATH,
-        #             UpdateArgsV1(code_path=None,
-        #                          config=ConfigV1(max_num_requests_to_remove=2,
-        #                                          request_timeout_blocks=10)),
-        #             id="update current config"),
-    ])
-def test_contract_update(initial_contract_path, update_args):
-    initial_contract = load_binary_file(initial_contract_path)
-    cluster = shared.start_cluster_with_mpc(2, 2, 1, initial_contract)
-    # assert correct contract is deployed
-    # todo: legacy cluster
-    cluster.assert_is_deployed(initial_contract)
-    cluster.init_contract(threshold=2)
-    # do some requests
-    cluster.send_and_await_signature_requests(2,
-                                              add_gas=150 * TGAS,
-                                              add_deposit=10)
-    # propose v1
-    cluster.propose_update(update_args.borsh_serialize())
+from common_lib.contracts import COMPILED_CONTRACT_PATH, MIGRATE_CURRENT_CONTRACT_PATH, V1_0_1_CONTRACT_PATH, UpdateArgsV1
+
+
+def deploy_and_init_v2():
+    cluster, mpc_nodes = shared.start_cluster_with_mpc(
+        2, 4, 1, contracts.load_mpc_contract())
+    cluster.init_cluster(participants=mpc_nodes[:2], threshold=2)
+    cluster.contract_state().print()
+    cluster.send_and_await_signature_requests(1)
+    public_key = cluster.contract_state().keyset().keyset[0].key
+    return cluster, mpc_nodes, public_key
+
+
+def get_participants_from_near_config():
+    # Get the participant set from the mpc configs
+    dot_near = pathlib.Path.home() / '.near'
+    with open(pathlib.Path(dot_near / 'participants.json')) as file:
+        participants_config = yaml.load(file,
+                                        Loader=shared.SafeLoaderIgnoreUnknown)
+
+    participants_map = {}
+    account_to_participant_id = {}
+    for i, p in enumerate(participants_config['participants']):
+        near_account = p['near_account_id']
+        my_pk = p['p2p_public_key']
+        my_addr = p['address']
+        my_port = p['port']
+
+        participants_map[near_account] = {
+            "account_id":
+            near_account,
+            "cipher_pk": [
```

Comment:
always funny to see a list of zeros in our integration tests :joy:

nit: [0] * 32